File: spec/factories/player.rb
FactoryBot.define do
  factory :player do
    name { Faker::Name.unique.name }
    birthdate { Faker::Date.birthday(min_age: 15, max_age: 95) }
    nationality {Faker::Name.unique.name }
    position { Player::POSITIONS.sample }
    sequence(:number) { |n| n }
  end

end

File: angular/src/app/shared/models/player.ts
export interface PlayerParameters {
  id?: string;
  birthdate: Date;
  name: string;
  nationality: string;
  number: number;
  position: string;
}

export class Player {
  id?: string;
  birthdate: Date;
  name: string;
  nationality: string;
  number: number;
  position: string;

  constructor({
    id,
    birthdate,
    name,
    nationality,
    number,
    position,
  }: PlayerParameters) {
    this.id = id;
    this.birthdate = birthdate;
    this.name = name;
    this.nationality = nationality;
    this.number = number;
    this.position = position;
  }
}
